### PR TITLE
BaSP-MR-152: Redux auth config

### PR DESCRIPTION
**Pull request by victoriassol**
Added login, signup and logout thunks, reducers and actions to be implemented in the UI.
Added the token header to each resource's thunks so that it checks for authorization in every request.
Initialized firebase.

### CODE REVIEW

File: src/Firebase/index.js
Comment (francoax) [inline src/Firebase/index.js:1]: This file is not necessary because in the setup @santimenendez98 has already created

**Review by francoax — CHANGES_REQUESTED** [review]

File: src/Redux/Auth/thunks.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  signUpSuccess,
+  getAuthenticationPending,
+  getAuthenticationSuccess,
+  getAuthenticationError,
+  logoutSuccess,
+  logoutError,
+  logoutPending
+} from './actions';
+
+import { firebaseApp } from '../../helper/firebase';
```
Comment (WalterArielPavicich) [inline src/Redux/Auth/thunks.js]: change relative to absolute route

**Review by santimenendez98 — APPROVED** [review]
Nice!

**Review by francoax — APPROVED** [review]

File: src/Redux/Auth/thunks.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      return dispatch(loginError(error.toString()));
+    }
+  };
+};
+
+export const getAuth = (token) => {
+  return async (dispatch) => {
+    dispatch(getAuthenticationPending());
+    try {
+      const response = fetch(`${process.env.REACT_APP_API_URL}/api/auth/`, { headers: { token } });
```
Comment (WalterArielPavicich) [inline src/Redux/Auth/thunks.js]: await is missing

```suggestion
      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/auth/`, { headers: { token } });
```

**Pull request merged**